design/open/pull-request.md

```diff
@@ -6,23 +6,30 @@ The pull-request captures the merge operation we request to apply.
 ### Goals
 
 - Capture request to merge a branch or specific reference
-- Mechanism to review, discuss and approve changes before applying to a branch
+- Mechanism to review, discuss and merge changes before applying to a branch
 - Automation using actions can verify / provide feedback on pull-request
 
 ### How it will work
 
 Introduce a new resource called pull-request at the repository level.
-The pull-request will hold owner (the user who created the PR), assignee, reviewers the source branch/reference and the target branch.
-Each reviewer can set approved/reject (or clear his response) on the PR.
+The pull-request will hold owner (the user who created the PR), description, assignee, reviewers the source branch/reference and the target branch.
+Each reviewer can set approved/reject (or clear his response) on the PR and add comments to the PR discussion. Comments associated to a PR will a flat ordered list of posted comments on the PR level.
+New comments are appended to the end of the discussion and will include a time stamp, commenter and the text with the comment.
 PR can be merge operation is enabled when at least one reviewer approves or when no reviewers are assigned.
 PR can be closed at any time.
-PR will not be updated when new commits into the source branch.
+PR will not be updated with changes to the source or destination branches, like updates to objects, commits, merges, etc. 
 Merge PR will perform a merge operation and merge the PR as merged. If a conflict or an error occurs while merging, it will be returned as a response to the request.
 When a repository is deleted, all pull-requests will be deleted with it.
 
+Actions - no change in current support with pre/post merge. These actions will be triggered when the PR is merged.
+A new event called 'pull-request' will be triggered when is created (or by request). These action will be triggered if no conflict found on an internal commit that will be visible to the action in order to verify the PR data.
```

Comment:
What does `by request` mean? Can I rerun an action of a PR? Can I request a trigger for PR that did not run in the PR?

Reply:
A request to re-run the action as the target branch can hold/update over time. We will need to enable re-running the action as it verify the merge with the target at a point in time.